Training/python/2018v1/Training_v0p1.py

```diff
@@ -267,16 +290,38 @@ def compile_model(model, opt_name, learning_rate):
 
 def run_training(model, data_loader, to_profile, log_suffix):
 
-    gen_train = data_loader.get_generator(primary_set = True, return_weights = data_loader.use_weights)
-    gen_val = data_loader.get_generator(primary_set = False, return_weights = data_loader.use_weights)
-    input_shape, input_types = data_loader.get_input_config()
-
-    data_train = tf.data.Dataset.from_generator(
-        gen_train, output_types = input_types, output_shapes = input_shape
-        ).prefetch(tf.data.AUTOTUNE)
-    data_val = tf.data.Dataset.from_generator(
-        gen_val, output_types = input_types, output_shapes = input_shape
-        ).prefetch(tf.data.AUTOTUNE)
+    if data_loader.input_type == "tf":
+        total_batches = data_loader.n_batches + data_loader.n_batches_val
+        ds = tf.data.experimental.load(data_loader.tf_input_dir, compression="GZIP") # import dataset
+        if data_loader.rm_inner_from_outer:
+            my_ds = ds.map(rm_inner)
+        else: 
+            my_ds = ds
+        cell_locations = data_loader.cell_locations
+        active_features = data_loader.active_features
+        active = [] #list of elements to be kept
+        if "TauFlat" in active_features:
+            active.append(1)
+        if "inner" in cell_locations:
+            active.extend([2,3,4])
+        if "outer" in cell_locations:
+            active.extend([5,6,7])
```

Comment:
The order of the cell may not be static and depends on how we define it in config: https://github.com/cms-tau-pog/TauMLTools/blob/ed7b7b385219fa6a239bf972502cb8edf92754d1/Training/configs/training_v1.yaml#L72-L76
Better to read dynamically order of CellType from config or precompiled enume classes. Same corresponds to rm_inner function.

Reply:
@shedprog I've now added an argument tf_dataset_x_order, which will allow us to specify the order of the layers in the tf dataset we load, in case eg we disabled some when it was generated, and rm_inner and reshape_tensor now read the relevant indeces from this